x-pack/plugins/infra/server/kibana.index.ts

```diff
@@ -20,6 +20,42 @@ export const initServerWithKibana = (kbnServer: KbnServer) => {
 
   // Register a function with server to manage the collection of usage stats
   kbnServer.usage.collectorSet.register(UsageCollector.getUsageCollector(kbnServer));
+
+  // @ts-ignore
```

Comment:
Having to do a `@ts-ignore` here feels wrong. The HapiJS type definitions has `plugins` being of type `PluginProperties` which is an empty interface: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/1f04344d92116837def38f25dd14586e8def467d/types/hapi/index.d.ts#L3318-L3319

What if we modifying the `Server` definition to the following instead:

```
interface KibanaPluginProperties extends PluginProperties {
  xpack_main: any;
}

export interface KbnServer extends Server {
  usage: any;
  plugins: KibanaPluginProperties;
}
```

Reply:
To be honest, this is the first time that I've tried this, I'm really not sure if it's best practice or not, but it seems to work.

Reply:
I was torn on this. I see other plugins doing just that, but I'd really love to have a centralized plugin definition within x-pack. I'll make this change for the time being, and see if there's a way for us to consolidate these definitions later on

Reply:
For sure, we could make the xpack_main plugin export it's own "public interface type" that we'd use in place of `xpack_main: any;`. I'm also not opposed to doing:

```
export interface KbnServer extends Server {
  usage: any;
  plugins: any;
}
```

which still feels more proper than throwing a @ts-ignore on it.